### PR TITLE
Check parents for ImplicitNotFound message

### PR DESCRIPTION
**Pull request by som-snytt**

### CODE REVIEW

**Review by NthPortal — APPROVED** [review]

**Comment by som-snytt** [comment]
@NthPortal Thanks Nth, you're the best. I'll take a look at the failing tests... :(

File: src/reflect/scala/reflect/internal/Symbols.scala [hunk 1]
```diff
@@ -928,7 +928,10 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     def migrationMessage     = getAnnotation(MigrationAnnotationClass) flatMap { _.stringArg(0) }
     def migrationVersion     = getAnnotation(MigrationAnnotationClass) flatMap { _.stringArg(1) }
     def elisionLevel         = getAnnotation(ElidableMethodClass) flatMap { _.intArg(0) }
-    def implicitNotFoundMsg  = getAnnotation(ImplicitNotFoundClass) flatMap { _.stringArg(0) }
+    def implicitNotFoundMsg  =
+      getAnnotation(ImplicitNotFoundClass).orElse(
```
Comment (NthPortal) [inline src/reflect/scala/reflect/internal/Symbols.scala]: What I don't understand is that the failing tests are already using custom `implicitNotFound` messages, which presumably get returned by `getAnnotation(ImplicitNotFoundClass)`; if so, how is the `orElse` ever executed to produce a different result?

Regardless, I think it is worth adding a test for the correct message being used when there are multiple in the parent chain.

File: src/reflect/scala/reflect/internal/Symbols.scala [hunk 2]
```diff
@@ -928,7 +928,10 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
...
+        parentSymbolsIterator.flatMap(_.getAnnotation(ImplicitNotFoundClass)).nextOption()
```
Comment (kubukoz) [inline src/reflect/scala/reflect/internal/Symbols.scala]: It's probably worth documenting what the resolution order is.

File: src/reflect/scala/reflect/internal/Symbols.scala (hunk 1 above)
Comment (som-snytt) [reply src/reflect/scala/reflect/internal/Symbols.scala]: There's a newish mechanism for checking if a param is annotated directly, before checking the type. "Parents of the param" accounts for seeing the wrong annotation (for some reason).

**Review by lrytz — APPROVED** [review]

**Comment by lrytz** [comment]
Some discussion at https://github.com/fthomas/refined/issues/661

**Pull request merged**